### PR TITLE
fix: App crashes when value in quantity textview is erased (#624)

### LINKED ISSUE #624: App crashes when value in Quantity textview is erased in create new ticket form.
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    
### Bug Report    

**Actual Behaviour**

App crashes when value in Quantity edittext of New ticket form is erased

**Expected Behaviour**

App should not crash

**LogCat for the issue**

02-02 22:24:09.995 3281-3281/org.fossasia.eventyay E/MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
02-02 22:24:10.095 3281-3281/org.fossasia.eventyay E/MessageQueue-JNI: java.lang.NumberFormatException: Invalid long: ""
                                                                           at java.lang.Long.invalidLong(Long.java:124)
                                                                           at java.lang.Long.parseLong(Long.java:345)
                                                                           at java.lang.Long.parseLong(Long.java:321)
                                                                           at org.fossasia.openevent.app.common.app.binding.BindingAdapters.strToLong(BindingAdapters.java:45)
                                                                           at org.fossasia.openevent.app.databinding.TicketCreateFormBinding$5.onChange(TicketCreateFormBinding.java:184)
                                                                           at android.databinding.adapters.TextViewBindingAdapter$1.onTextChanged(TextViewBindingAdapter.java:367)
                                                                           at android.widget.TextView.sendOnTextChanged(TextView.java:7991)
                                                                           at android.widget.TextView.handleTextChanged(TextView.java:8053)
                                                                           at android.widget.TextView$ChangeWatcher.onTextChanged(TextView.java:10157)
                                                                           at android.text.SpannableStringBuilder.sendTextChanged(SpannableStringBuilder.java:1033)
                                                                           at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:559)
                                                                           at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:225)
                                                                           at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:224)
                                                                           at android.text.method.BaseKeyListener.backspaceOrForwardDelete(BaseKeyListener.java:106)
                                                                           at android.text.method.BaseKeyListener.backspace(BaseKeyListener.java:51)
                                                                           at android.text.method.BaseKeyListener.onKeyDown(BaseKeyListener.java:215)
                                                                           at android.text.method.NumberKeyListener.onKeyDown(NumberKeyListener.java:138)
                                                                           at android.widget.TextView.doKeyDown(TextView.java:6098)
                                                                           at android.widget.TextView.onKeyDown(TextView.java:5911)
                                                                           at android.view.KeyEvent.dispatch(KeyEvent.java:2640)
                                                                           at android.view.View.dispatchKeyEvent(View.java:9234)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.support.v4.widget.NestedScrollView.dispatchKeyEvent(NestedScrollView.java:547)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                           at com.android.internal.policy.PhoneWindow$DecorView.superDispatchKeyEvent(PhoneWindow.java:2395)
                                                                           at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1727)
                                                                           at android.app.Dialog.dispatchKeyEvent(Dialog.java:753)
                                                                           at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
                                                                           at android.support.v7.app.AppCompatDelegateImplBase$AppCompatWindowCallbackBase.dispatchKeyEvent(AppCompatDelegateImplBase.java:316)
                                                                           at com.android.internal.policy.PhoneWindow$DecorView.dispatchKeyEvent(PhoneWindow.java:2310)
                                                                           at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4127)
                                                                           at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4089)
                                                                           at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3642)
                                                                           at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3695)
                                                                           at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3661)
                                                                           at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3787)
                                                                           at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3669)
                                                                           at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3844)
                                                                           at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3642)
                                                                           at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3695)
                                                                           at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3661)
                                                                       	at android.view.ViewRootImpl$InputSta
02-02 22:24:10.096 3281-3281/org.fossasia.eventyay D/AndroidRuntime: Shutting down VM
                                                                     
                                                                     
                                                                     --------- beginning of crash
02-02 22:24:10.097 3281-3281/org.fossasia.eventyay E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     Process: org.fossasia.eventyay, PID: 3281
                                                                     java.lang.NumberFormatException: Invalid long: ""
                                                                         at java.lang.Long.invalidLong(Long.java:124)
                                                                         at java.lang.Long.parseLong(Long.java:345)
                                                                         at java.lang.Long.parseLong(Long.java:321)
                                                                         at org.fossasia.openevent.app.common.app.binding.BindingAdapters.strToLong(BindingAdapters.java:45)
                                                                         at org.fossasia.openevent.app.databinding.TicketCreateFormBinding$5.onChange(TicketCreateFormBinding.java:184)
                                                                         at android.databinding.adapters.TextViewBindingAdapter$1.onTextChanged(TextViewBindingAdapter.java:367)
                                                                         at android.widget.TextView.sendOnTextChanged(TextView.java:7991)
                                                                         at android.widget.TextView.handleTextChanged(TextView.java:8053)
                                                                         at android.widget.TextView$ChangeWatcher.onTextChanged(TextView.java:10157)
                                                                         at android.text.SpannableStringBuilder.sendTextChanged(SpannableStringBuilder.java:1033)
                                                                         at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:559)
                                                                         at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:225)
                                                                         at android.text.SpannableStringBuilder.delete(SpannableStringBuilder.java:224)
                                                                         at android.text.method.BaseKeyListener.backspaceOrForwardDelete(BaseKeyListener.java:106)
                                                                         at android.text.method.BaseKeyListener.backspace(BaseKeyListener.java:51)
                                                                         at android.text.method.BaseKeyListener.onKeyDown(BaseKeyListener.java:215)
                                                                         at android.text.method.NumberKeyListener.onKeyDown(NumberKeyListener.java:138)
                                                                         at android.widget.TextView.doKeyDown(TextView.java:6098)
                                                                         at android.widget.TextView.onKeyDown(TextView.java:5911)
                                                                         at android.view.KeyEvent.dispatch(KeyEvent.java:2640)
                                                                         at android.view.View.dispatchKeyEvent(View.java:9234)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.support.v4.widget.NestedScrollView.dispatchKeyEvent(NestedScrollView.java:547)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1640)
                                                                         at com.android.internal.policy.PhoneWindow$DecorView.superDispatchKeyEvent(PhoneWindow.java:2395)
                                                                         at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1727)
                                                                         at android.app.Dialog.dispatchKeyEvent(Dialog.java:753)
                                                                         at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
                                                                         at android.support.v7.app.AppCompatDelegateImplBase$AppCompatWindowCallbackBase.dispatchKeyEvent(AppCompatDelegateImplBase.java:316)
                                                                         at com.android.internal.policy.PhoneWindow$DecorView.dispatchKeyEvent(PhoneWindow.java:2310)
                                                                         at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4127)
                                                                         at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4089)
                                                                         at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3642)
                                                                         at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3695)
                                                                         at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3661)
                                                                         at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3787)
                                                                         at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3669)
                                                                         at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3844)
                                                                         at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3642)
                                                                         at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3695)
                                                                     	at android.view.ViewRootImpl$InputStage.forward(    

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AkshayCHD**
Fixes #624

Changes: makes the default quantity zero if the quantity field in create new ticket form is left blank 
which prevents the app from crashing when the value is erased.

### CODE REVIEW

**Comment by AkshayCHD** [comment]
@iamareebjamal sir can you help me out why my travis-ci check is failing

**Comment by AkshayCHD** [comment]
thanks for the advice @Masquerade0097 , but now my Circle ci test failed, can you help again

**Comment by iamareebjamal** [comment]
@AkshayCHD CircleCI is optional. It is also sometimes unreliable. I'm trying to fix it for a few days. Only check travis

File: app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public static String floatToStr(Float value) {
     }
 
     public static Long strToLong(String value) {
-        return value == null ?  null : Long.parseLong(value);
+        return (("").equals(value)) ? 0 : Long.parseLong(value);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: Use TextUtils.isEmpty

Secondly, we don't want 0 to go to server if user has not entered anything, we want null

**Comment by AkshayCHD** [comment]
@iamareebjamal sir please review

**Comment by iamareebjamal** [comment]
@AkshayCHD Please fix travis build

**Comment by AkshayCHD** [comment]
@iamareebjamal how can i track why my travis ci test is failing

**Comment by iamareebjamal** [comment]
https://travis-ci.org/fossasia/open-event-orga-app/builds/337004381?utm_source=github_status&utm_medium=notification

**Comment by iamareebjamal** [comment]
@AkshayCHD There are PMD violations in your build. Please check and correct them and then send PR. Thanks

**Comment by AkshayCHD** [comment]
@iamareebjamal removed the travis issue

File: app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java [hunk 2]
```diff
@@ -42,7 +42,10 @@ public static String floatToStr(Float value) {
     }
 
     public static Long strToLong(String value) {
-        return value == null ?  null : Long.parseLong(value);
+        if (value.isEmpty())
+            return null;
+        else
+            return Long.parseLong(value);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: Use `StringUtils.isEmpty(value) ? null : Long.parseLong(value)`

**Comment by AkshayCHD** [comment]
@iamareebjamal when I use ```StringUtils.isEmpty(value) ? null : Long.parseLong(value)``` i get PMD violation 
Assigning an Object to null is a code smell.  Consider refactoring
 but when i do it without ternary operator it shows no such issue.

**Comment by iamareebjamal** [comment]
You should directly return and not assign

**Comment by AkshayCHD** [comment]
@iamareebjamal i placed the code directly in the return statement, i didn't assign

**Comment by iamareebjamal** [comment]
Then you may add a @SuppressWarnings annotation for PMD violation

**Comment by AkshayCHD** [comment]
@iamareebjamal please review, hopefully the last time :sweat_smile:

File: app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 import android.widget.ImageView;
 
 import com.mikhaellopez.circularprogressbar.CircularProgressBar;
+import com.raizlabs.android.dbflow.StringUtils;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: Use the app's StringUtils, not dbflow's

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java [hunk 4]
```diff
@@ -12,13 +12,15 @@
 import android.support.v4.content.ContextCompat;
 import android.support.v4.graphics.drawable.DrawableCompat;
 import android.support.v4.view.ViewCompat;
+import android.text.TextUtils;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: Not the android TextUtils, use the app one

Comment (AkshayCHD) [reply app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: @iamareebjamal there is only android TextUtils available and no TextUtils function in org.fossasia.openevent.app.common.utils.core. there is only string utils available and StringUtils has no function isEmpty()

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: https://github.com/fossasia/open-event-orga-app/blob/development/app/src/main/java/org/fossasia/openevent/app/common/utils/core/Utils.java#L28

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java [hunk 5]
```diff
@@ -45,9 +46,9 @@ public static String floatToStr(Float value) {
     public static String doubleToStr(Double value) {
         return value == null ? "" : String.valueOf(value);
     }
-
+    @SuppressWarnings("PMD")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/app/binding/BindingAdapters.java]: Please add a line break after function declaration

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

**Pull request merged**